pvanalytics/features/orientation.py

```diff
@@ -0,0 +1,185 @@
+                     peak_min=None):
+    # If there are at least `min_hours` of data in `day` and the
+    # maximum for the day is greater than `peak_min` then `fitfunc` is
+    # applied to fit a curve to the data. `fitfunc` must be a function
+    # that takes a Series and returns the :math:`r^2` for a curve fit.
+    # If the two conditions are not met then `default` is returned and
+    # no curve fitting is performed.
+    high_enough = True
+    if peak_min is not None:
+        high_enough = day.max() > peak_min
+    if (_hours(day, freq) > min_hours) and high_enough:
+        return fitfunc(day)
+    return default
+
+
+def _freqstr_to_hours(freq):
+    # Convert pandas freqstr to hours (as a float)
+    if freq.isalpha():
+        freq = '1' + freq
+    return pd.to_timedelta(freq).seconds / 60
+
+
+def _hours(data, freq):
+    # Return the number of hours in `data` with data where timestamp
+    # spacing is given by `freq`.
+    return data.count() * _freqstr_to_hours(freq)
+
+
+def _group_by_day(data):
+    # Group data by timezone localized date.
+    #
+    # We use this function (rather than `data.resample('D')`) because
+    # Resampler.apply() makes the data tz-naive when passed to the
+    # function being applied. This causes the curve fitting functions
+    # to fail when the minute-of-the-day is used as the x-coordinate
+    # since removing timezone information can cause data for a day to
+    # span two dates.
+    #
+    # The date needs to be localized in the same timezone as
+    # data.index so we can easily reindex the series with the original
+    # index at a later time.
+    return data.groupby(
+        pd.to_datetime(data.index.date).tz_localize(data.index.tz)
+    )
+
+
+def tracking(power_or_irradiance, daytime, correlation_min=0.94,
+             fixed_max=0.96, min_hours=5, peak_min=None,
+             late_morning='08:45', early_afternoon='17:15'):
+    """Flag days where the data matches the profile of a tracking PV system.
+
+    Tracking days are identified by fitting a restricted quartic to
+    the data for each day. If the :math:`r^2` for the fit is greater
+    than `correlation_min` and the :math:`r^2` for a quadratic fit is
+    less than `fixed_max` then the data on that day is all marked
+    True. Data on days where the tracker is stuck, or there is
+    substantial variability in the data (i.e. caused by cloudiness) is
+    marked False.
+
+    Parameters
+    ----------
+    power_or_irradiance : Series
+        Timezone localized series of power or irradiance measurements.
+    daytime : Series
+        Boolean series with True for times that are during the
+        day. For best results this mask should exclude early morning
+        and evening as well as night. Morning and evening may have
+        problems with shadows that interfere with curve fitting, but
+        do not necessarily indicate that the day was not sunny.
```

Comment:
confused by the reference to "sunny" days here - is the returned series True only when both sunny and looks-like-tracking?